app/src/main/java/net/bible/service/db/readingplan/ReadingPlanDBAdapter.kt

```diff
@@ -0,0 +1,770 @@
+import java.util.Date
+import kotlin.collections.ArrayList
+
+/**
+ * @author Timmy Braun [tim.bze at gmail dot com] (2/14/2019)
+ */
+class ReadingPlanDBAdapter {
+    companion object {
+        const val TAG = "ReadingPlanDBAdapter"
+        const val INCLUSIVE_VERSIFICATION = SystemNRSVA.V11N_NAME
+        val app = BibleApplication.application
+        val dbHelper: CommonDatabaseHelper = CommonDatabaseHelper.getInstance()
+
+        private val dbWritable = dbHelper.writableDatabase
+        private val dbReadable = dbHelper.readableDatabase
+
+        private val dateBasedFormatMonthDay = SimpleDateFormat("MMM-d")
+        val dateBasedFormatWithYear = SimpleDateFormat("MMM-d/yyyy")
+
+    }
+
+    var metaCurrentActiveReadingPlanID: Int?
+        get() {
+            val projection = arrayOf(BaseColumns._ID)
+            val selection = "${ReadingPlanMeta.COLUMN_LAST_USED_PLAN}=?"
+            val selectionArgs = arrayOf(DB_TRUE_VALUE)
+            val cursor = dbReadable.query(
+                ReadingPlanMeta.TABLE_NAME,
+                projection,
+                selection,
+                selectionArgs,
+                null,
+                null,
+                null)
+            val result =
+                if (cursor.moveToFirst())
+                    cursor.getInt(0)
+                else null
+
+            cursor.close()
+            return result
+        }
+        set(newValue) {
+            if (newValue != null && newValue != 0) {
+                val setFalseValues = ContentValues().apply {
+                    put(ReadingPlanMeta.COLUMN_LAST_USED_PLAN, DB_FALSE_VALUE)
+                }
+                val setTheTrueValue = ContentValues().apply {
+                    put(ReadingPlanMeta.COLUMN_LAST_USED_PLAN, DB_TRUE_VALUE)
+                }
+                val whereClause = "${ReadingPlanMeta.COLUMN_ID}=?"
+                val whereArgs = arrayOf(newValue.toString())
+                // Set all plans to false. ONLY 1 plan is the Current Active Plan
+                dbWritable.update(
+                    ReadingPlanMeta.TABLE_NAME,
+                    setFalseValues,
+                    null,
+                    null)
+                // Set the 1 Current Plan to Active
+                dbWritable.update(
+                    ReadingPlanMeta.TABLE_NAME,
+                    setTheTrueValue,
+                    whereClause,
+                    whereArgs)
+            }
+        }
+
+    val currentPlanShortCode: String get() =
+        StringUtils.left(getMetaFileNameFromId(metaCurrentActiveReadingPlanID ?: 0),8)
+    val currentDayAndNumberDescription: String get() =
+        app.getString(
+            R.string.rdg_plan_day,
+            getMetaCurrentDayNumber(metaCurrentActiveReadingPlanID ?: 0).toString()
+        )
+    /** Will always get or set the current day number from the active reading plan
+     * @return Null result means that it is a date-based plan and there are no readings for today.
```

Comment:
this can't return null

Reply:
Duh, I guess that's a comment from when I still did return null :)
Fixed!